files/class-vip-filesystem-stream-wrapper.php

```diff
@@ -854,13 +854,41 @@ public function validate( $path, $mode ) {
 	 * @since   1.0.0
 	 * @access  protected
 	 * @param   string    $message  Debug message to be logged
+	 * @param   bool      $force Whether to force debug
 	 */
-	protected function debug( $message ) {
-		if ( ! $this->debug_mode ) {
+	protected function debug( $message, $force = false ) {
+		if ( ! ( $this->debug_mode || $force ) ) {
 			return;
 		}
 
-		// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
-		error_log( sprintf( '%s (%s)', $message, wp_debug_backtrace_summary() ) );
+		$trace = $this->backtrace_fmt();
+
+		\Automattic\VIP\Logstash\log2logstash(
+			[
+				'severity' => 'info',
+				'feature'  => 'stream_wrapper_audit_' . $trace[1]['function'],
+				'message'  => "File op {$trace[1]['function']}: " . $message,
+				'extra'    => [
+					'trace' => $this->backtrace_fmt(),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Format the debug backtrace to be a bit more readable .
+	 *
+	 * @return array
+	 */
+	private function backtrace_fmt() {
+		$trace = debug_backtrace( false, 30 ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_debug_backtrace
```

Comment:
`false` => `0`, as the first argument is [`int`](https://www.php.net/manual/en/function.debug-backtrace.php)